Standardize retriever span outputs to a list of MLflow `Document`s (#13242)
mlflow/mlflow: docs/source/llms/tracing/tracing-schema.rst

Schema for specific span types
------------------------------

MLflow has a set of 10 predefined types of spans (see :py:class:`mlflow.entities.SpanType`), and
certain span types have properties that are required in order to enable additional functionality
within the UI and downstream tasks such as evaluation.

Retriever Spans
^^^^^^^^^^^^^^^

The ``RETRIEVER`` span type is used for operations involving retrieving data from a data store (for example, querying 
documents from a vector store). The ``RETRIEVER`` span type has the following schema:

.. list-table::
    :widths: 20 40 40
    :header-rows: 1
    :class: wrap-table

    * - **Property**
      - **Description**
      - **Note**

    * - **Input**
      - There are no restrictions on the span inputs
      -
    
    * - **Output**
      - The output must be of type ``List[`` :py:class:`mlflow.entities.Document` ``]``, or a dict matching the structure of the dataclass\*. 
        The dataclass contains the following properties:

        * **id** (``Optional[str]``) - An optional unique identifier for the document.
        * **page_content** (``str``) - The text content of the document.

>         * **metadata** (``Optional[Dict[str,any]]``) - The metadata associated with the document. There are two important metadata keys that are used by the UI and evaluation metrics: 
> 

Reviewer: ```suggestion
        * **metadata** (``Optional[Dict[str,any]]``) - The metadata associated with the document. There are two important metadata keys that are reserved for the MLflow UI and evaluation metrics: 
```

>         * **metadata** (``Optional[Dict[str,any]]``) - The metadata associated with the document. There are two important metadata keys that are reserved for the MLflow and evaluation metrics: 

Author: ```suggestion
        * **metadata** (``Optional[Dict[str,any]]``) - The metadata associated with the document. There are two important metadata keys that are reserved for the MLflow UI and evaluation metrics: 
```
